packages/dd-trace/src/tracer.js

```diff
@@ -123,6 +123,10 @@ class DatadogTracer extends Tracer {
 <meta name="dd-trace-id" content="${traceId}" />\
 <meta name="dd-trace-time" content="${traceTime}" />`
   }
+
+  flush () {
+    return new Promise(resolve => this._exporter._writer.flush(resolve))
```

Comment:
Node has issues with promises when combined with `async_hooks` so we try to avoid them whenever possible in the tracer. Can this take a callback instead and then you could promisify it externally?

Reply:
Done 👍